src/Microsoft.IdentityModel.JsonWebTokens/Json/JsonClaimSet.cs

```diff
@@ -24,14 +25,44 @@ internal class JsonClaimSet
 
         internal object _claimsLock = new();
         internal readonly Dictionary<string, object> _jsonClaims;
+
+#if NET8_0_OR_GREATER
```

Comment:
Aren't there earlier versions that would support this?
```suggestion
#if NET5_0_OR_GREATER
```

Reply:
Certain APIs on the Utf8JsonReader are available only on net8 target.